modules/programs/calcurse.nix

```diff
@@ -0,0 +1,191 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.calcurse;
+
+  formatLine = n: v:
+    let
+      formatValue = v:
+        if isBool v then (if v then "yes" else "no")
+        else toString v;
+    in
+      "${lib.toLower n}=${formatValue v}";
+
+in
+
+{
+  options.programs.calcurse = {
+    enable = mkEnableOption "calcurse";
+
+    general = {
+      autoSave = mkOption {
+        type = types.bool;
+        default = true;
+        description = "This option allows to automatically save the user’s data (if set to yes) when quitting. warning: No data will be automatically saved if general.autosave is set to no. This means the user must press S (for saving) in order to retrieve its modifications.";
```

Comment:
Line width should be limited to 80 or so characters.

Also, does the license of calcurse allow copying the descriptions directly into a MIT licensed project?

Also, consider only defining the most important or commonly used program options and instead add an `extraConfig` option for those users that need the remaining ones.

Reply:
I removed fomat and daemon, since they're easier to set with extraConfig. The calcurse license is [BSD 2](https://github.com/lfos/calcurse/blob/pu/COPYING), so I think it'd be okay to copy.